src/providers/LedgerProvider.js

```diff
@@ -23,33 +23,35 @@ export default class LedgerProvider extends WalletProvider {
   }
 
   async createTransport () {
-    if (!LedgerProvider.transport) {
+    if (!this._transport) {
       debug('creating ledger transport')
-      LedgerProvider.transport = await Transport.create()
+      this._transport = await Transport.create()
       debug('ledger transport created')
-      LedgerProvider.transport.on('disconnect', () => {
+      this._transport.on('disconnect', () => {
         debug('ledger disconnected')
         this._appInstance = null
-        LedgerProvider.transport = null
+        this._transport = null
       })
     }
   }
 
   errorProxy (target, func) {
     const method = target[func]
+    const ctx = this
     if (Object.getOwnPropertyNames(target).includes(func) && typeof method === 'function') {
       return async (...args) => {
         debug(`calling "${func}" on ledger object`, args)
 
         try {
           const result = await method.bind(target)(...args)
-
           debug(`result from "${func}" on ledger object`, result)
-
           return result
         } catch (e) {
-          const { name, ...errorNoName } = e
-          throw new WalletError(e.toString(), errorNoName)
+          // const { name, ...errorNoName } = e
+          ctx._transport = null
+          ctx._appInstance = null
+          // throw new WalletError(e.toString(), errorNoName)
```

Comment:
Is it okay to return null instead of throwing an error? This will simply return a null value if any ledger function fails, and it will make checking for null mandatory everywhere it is used, right?

Reply:
Ageed. The purpose of this was so that `isWalletAvailable` returns false instead of an error? I think we discussed still throwing the error but having the first line in `isWalletAvailable` catch it and then return false.

Reply:
Agree and throwing again and catching inside of `isWalletAvailable`